### PR TITLE
Scripts role

### PR DESCRIPTION
**Pull request by jburel**
Add a new role to install dependencies required to run the official scripts
figure to pdf should be considered as an official script
2 files so it can be used on centos and ubuntu
test added

To test:

 * centos: Run ``vagrant up scriptsutils`` 
 * ubuntu: 
   * change ``node.vm.box = "centos/7"`` to ``node.vm.box = "ubuntu/trusty64"`` (I have not yet tried to pass it as a parameter!)
   * Run ``vagrant up scriptsutils`` 

Unfortunately, no official "mencoder"
If someone finds a better repo, please let me know

### CODE REVIEW

**Comment by jburel** [comment]
just realise a commit is missing for mencoder redhat file

File: ansible/roles/scripts-utils/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+# tasks file for roles/scripts-deps
+
+- include: redhat.yml
+  when: ansible_os_family == 'RedHat'
+  tags: package
```
Comment (manics) [inline ansible/roles/scripts-utils/tasks/main.yml]: @joshmoore Is this a standard tag in our repo?

File: ansible/roles/scripts-utils/tasks/redhat.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - python-reportlab
+    - python-markdown
+
+# dependency for make movie
+- name: Import key to install mencoder
+  sudo: yes
+  rpm_key: state=present key=http://li.nux.ro/download/nux/RPM-GPG-KEY-nux.ro
```
Comment (manics) [inline ansible/roles/scripts-utils/tasks/redhat.yml]: Can you change this to long-form YAML?

File: ansible/test.yml [hunk 3]
```diff
@@ -114,3 +114,18 @@
...
+
+- hosts: scriptsutils
+  roles:
+  - role: scripts-utils
+
+  tasks:
+  - command: python -c 'import reportlab'
+  - command: python -c 'import markdown'
+  - command: python -c 'import os.path; print os.path.exists("/usr/bin/mencoder")'
+    register: value
```
Comment (manics) [inline ansible/test.yml]: This works, but it'd be neater to test for the existence of the file using plain Ansible: Scroll down to

> Should you feel the need to test for existence of files that are not declaratively set by your Ansible configuration, the ‘stat’ module is a great choice:

on http://docs.ansible.com/ansible/test_strategies.html#modules-that-are-useful-for-testing

Alternatively you could just omit all these tests, it's a fairly simple role which installs stuff so if something goes wrong you'd expect to get an error whilst the role was running.

Comment (jburel) [inline ansible/test.yml]: I understood that we need to have tests even for small roles

File: ansible/roles/scripts-utils/tasks/main.yml (hunk 1 above)
Comment (manics) [inline ansible/roles/scripts-utils/tasks/main.yml]: @jburel Can you remove `tags: ` then?

**Comment by jburel** [comment]
@manics: all your comments should now be addressed. Thanks

File: ansible/Vagrantfile [hunk 4]
```diff
@@ -37,6 +37,7 @@ Vagrant.configure(2) do |config|
     "redis",
     "samba",
     "selinux",
+    "omeroscriptsdependencies"
```
Comment (manics) [inline ansible/Vagrantfile]: Very nitpicky comment: can you order this list alphabetically?

File: ansible/test.yml [hunk 5]
```diff
@@ -130,3 +130,18 @@
       state: directory
   # Setup a port-forward to 8000 and you should be able to login as
   # user:user password:ome
+
+- hosts: omeroscriptsdependencies, omeroscriptsdependencies-debian
+  roles:
+  - role: omero-scripts-dependencies
+
+  tasks:
+  - stat: path=/usr/bin/mencoder
```
Comment (manics) [inline ansible/test.yml]: YAML long form

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#1086](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1086/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1086/consoleText) for more details.
Possible conflicts:
  - PR #157 manics 'omero user role'
    - ansible/Vagrantfile

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [MANAGEMENT_TOOLS-merge#1087](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1087/). See the [console output](https://ci.openmicroscopy.org/job/MANAGEMENT_TOOLS-merge/1087/consoleText) for more details.
Possible conflicts:
  - PR #157 manics 'omero user role'
    - ansible/Vagrantfile

**Comment by jburel** [comment]
@manics: requested changes are now in

**Comment by manics** [comment]
@jburel Thanks, can you rebase to fix the Vagrantfile conflict, then this should be good to merge.

File: ansible/Vagrantfile [hunk 6]
```diff
@@ -47,6 +48,13 @@ Vagrant.configure(2) do |config|
   end
 
   [
+    "omeroscriptsdependencies-debian"
+  ].each do |server|
+    config.vm.define "#{server}" do |node|
+      node.vm.box = "ubuntu/trusty64"
+      node.vm.provision "ansible" do |ansible|
+        ansible.playbook = "test.yml"
```
Comment (manics) [inline ansible/Vagrantfile]: The merge has gone wrong, you'll need to add three `end`s in to close the three loops.

Comment (jburel) [inline ansible/Vagrantfile]: crap. That will teach me to do it just before the seminar!!

**Comment by jburel** [comment]
done sorry about that

**Comment by manics** [comment]
This all looks good, but `- name: Import key to install mencoder` on centos/7 is failing.
```
[vagrant@localhost ~]$ curl -o /dev/null http://li.nux.ro/download/nux/RPM-GPG-KEY-nux.ro
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
 82  3109   82  2578    0     0  14775      0 --:--:-- --:--:-- --:--:-- 14816
curl: (56) Recv failure: Connection reset by peer
```
This worked fine last week.

@openmicroscopy/devops: should we merge this? Or do you want to hold off?

**Comment by jburel** [comment]
@manics: That's what I was mentioning today, not using official repo is not ideal
I have seen this error one evening from home too
I will check if I can find another repo

**Comment by joshmoore** [comment]
:+1: to that @jburel. Also, the lack of https is a bit worrisome.

**Comment by jburel** [comment]
removed the mencoder installation in redhat file 
so this PR can be included.
I will re-add if/when I find a suitable alternative

**Comment by joshmoore** [comment]
mencoder working for debian; not for centos. (Test will need updating). Otherwise, probably best to get in and then we can follow up later.

File: ansible/test.yml [hunk 7]
```diff
@@ -130,3 +130,19 @@
...
+  - role: omero-scripts-dependencies
+
+  tasks:
+  - stat: 
+      path: "/usr/bin/mencoder"
+    register: p
+  - command: python -c 'import reportlab'
+  - command: python -c 'import markdown'
+
+   - assert:
```
Comment (manics) [inline ansible/test.yml]: Syntax error, you need to un-indent by one space.

File: ansible/roles/omero-scripts-dependencies/README.tmd [hunk 8]
```diff
@@ -0,0 +1,10 @@
+OMERO Scripts Dependencies
+==========================
+
+Install the dependencies required by the OMERO scripts
+
+
```
Comment (manics) [inline ansible/roles/omero-scripts-dependencies/README.tmd]: Can you add a note that this role is still incomplete for CentOS7? Also the file extension is `.tmd`, shouldn't it be `.md`?

**Comment by jburel** [comment]
@manics done

**Pull request merged**